purldb-toolkit/README.rst

```diff
@@ -20,84 +25,541 @@ Use this command to get basic help::
+                "exists": null,
+                "message": "The provided PackageURL is valid, but `check_existence` is not supported for this package type."
+            }
+        ]
+    }
+
+
+**Submit multiple PURLs using a .txt file:**
+
+.. code-block:: none
+
+    purlcli validate --file <path/to/output.txt> --output <path/to/output.json>
+
+*Sample input.txt:*
+
+.. code-block:: console
+
+    pkg:npm/canonical-path@1.0.0
+    pkg:nginx/nginx@0.8.9
+
+
+Notes
+#######
+
+``validate`` calls the ``public.purldb.io/api/validate/`` endpoint.
+
+
+----
+
+
+'versions' -- collect package versions for a PURL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: none
 
-- Collect package versions for a PURL::
-      
     $ purlcli versions  --help
     Usage: purlcli versions [OPTIONS]
-    
+
       Given one or more PURLs, return a list of all known versions for each PURL.
-    
-      Version information is not needed in submitted PURLs and if included will be
-      removed before processing.
-    
+
     Options:
       --purl TEXT        PackageURL or PURL.
       --output FILENAME  Write versions output as JSON to FILE.  [required]
       --file FILENAME    Read a list of PURLs from a FILE, one per line.
       --help             Show this message and exit.
 
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli versions --purl pkg:npm/canonical-path --purl pkg:nginx/nginx --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
+
+    {
+        "headers": [
+            {
+                "tool_name": "purlcli",
+                "tool_version": "0.2.0",
+                "options": {
+                    "command": "versions",
+                    "--purl": [
+                        "pkg:npm/canonical-path",
+                        "pkg:nginx/nginx"
+                    ],
+                    "--file": null,
+                    "--output": "<path/to/output.json>"
+                },
+                "errors": [],
+                "warnings": [
+                    "'pkg:nginx/nginx' not supported with `versions` command"
+                ]
+            }
+        ],
+        "packages": [
+            {
+                "purl": "pkg:npm/canonical-path@0.0.1",
+                "version": "0.0.1",
+                "release_date": "2013-12-19"
+            },
+            {
+                "purl": "pkg:npm/canonical-path@0.0.2",
+                "version": "0.0.2",
+                "release_date": "2013-12-19"
+            },
+            {
+                "purl": "pkg:npm/canonical-path@1.0.0",
+                "version": "1.0.0",
+                "release_date": "2018-10-24"
+            }
+        ]
+    }
+
 
-- Collect package metadata for a PURL::
+Notes
+#######
+
+``versions`` calls ``versions()`` from `fetchcode/package_versions.py <https://github.com/nexB/fetchcode/blob/master/src/fetchcode/package_versions.py>`__.
```

Comment:
Maybe we need better documentation on fetchcode features and what ecosystems are supported (like at https://github.com/nexB/fetchcode/blob/master/src/fetchcode/package_versions.py#L39 but this is generated programmatically in the docs, see an example in https://github.com/nexB/scancode-toolkit/commit/67256a1d5312d2b9a09e99ca3f53f7bd3c658ccc) This should be in fetchcode docs and linked here. This is probably out of scope but something to consider for later

Reply:
Basically we should avoid linking to code directly, as this is bad practice, and also the code might change. Easier to link to a section in another docs (via intersphinx)

Reply:
I removed the direct links to code as you requested -- I'd added these (for `versions` and `metadata`, which rely on `fetchcode/package_versions.py` and `fetchcode/package.py`, respectively) only because I could find no documentation or similar references to link to.